### ISSUE TITLE
How to get the audio microphone to work?

### ISSUE THREAD

**Issue by Therealjosephchrzempiec** [networked-aframe/networked-aframe#352]
Hello, I know this has been asked but I'm not understanding how to get the audio microphone to work on a local server?

Joseph

**Comment by vincentfretin** [comment]
There is no difference between hosting on a local server or on glitch. You may need a [turn server](https://github.com/networked-aframe/networked-aframe/blob/master/docs/hosting-networked-aframe-on-a-server.md#extra-steps-for-webrtc) for a local server if you want external users of your LAN to connect to your local server though.

You have a live example of using the mic with the easyrtc adapter here
https://naf-examples.glitch.me/basic-audio.html
You can look at the code in [basic-audio.html](https://github.com/networked-aframe/networked-aframe/blob/35a81751c50c3d56adaef8d60f0e26e53f8be5dc/examples/basic-audio.html) and compare it with the [basic.html](https://github.com/networked-aframe/networked-aframe/blob/35a81751c50c3d56adaef8d60f0e26e53f8be5dc/examples/basic.html) example to see what needs to be changed.

**Comment by Therealjosephchrzempiec** [comment]
Hello Vince, I have tried it and I can not get microphone to show up. I know my microphone works because I can skype and audio message people with no problem.

Joseph

**Comment by vincentfretin** [comment]
Look at the mic permission on the url bar, revoke it and refresh?

**Comment by vincentfretin** [comment]
When you accept the mic permission in the browser, you can select the mic, it will use the selected mic for this session. But the browser won't save the preferred mic I think? (I can't test because I don't have a working external mic, I use the default internal mic), so next time you load the page, it won't ask the mic permission again and it will probably use the wrong default mic. Maybe this is your issue?

To fix this issue we need #300 so we can save the preferred mic to localStorage for example `localStorage.setItem("lastUsedMicDeviceId", micDeviceId);`
and select the correct mic the next session like this:
```
const lastUsedMicDeviceId = localStorage.getItem("lastUsedMicDeviceId");
const constraints = {
  audio: {
    echoCancellation: true,
    noiseSuppression: true,
    autoGainControl: true,
    deviceId: { exact: lastUsedMicDeviceId },
  },
};
const newStream = await navigator.mediaDevices.getUserMedia(constraints);
....
```
